src/Listener/JsonApiListener.php

```diff
@@ -1078,7 +1079,7 @@ protected function _getSingleEntity(Subject $subject): ?EntityInterface
             return $subject->entities->first();
         }
 
-        return $subject->entity;
+        return $subject->get('query') ? $subject->get('query')->first() : $subject->get('entity');
```

Comment:
Please double-check this because `Subject::get()` does not implement a `null` pattern; but without the check `JsonApiListenerTest::testGetSingleEntity` fails.

Reply:
The possibility of `entity` not being set can be handled using `return $subject->entity ?? null;`

Reply:
Why would you use the query if the entity is already available?

Reply:
Sometimes (I don't know all the cases) the entity must come from the query as it isn't set previously.

Reply:
Still as per the current code `$subject->entity` should be checked first and then use the query if entity is not available.